File: 3/python/5/gyak05/stack.py
#Definiáljuk a Stack adatszerkezetet
#Emlékeztető: Stack műveletek: push, pop, peek, isEmpty, size

class Stack():
    def __init__(self):
        self.stack = []

    def push(self,item):
        self.stack.append(item)

    def pop(self):
        if self.isEmpty():
            return "Empty stack!"
        val = self.stack[-1]
        del self.stack[-1]
        return val
    
    def peek(self):
        if self.isEmpty():
            return "Empty stack!"
        return self.stack[-1]
    
    def isEmpty(self):
        return len(self.stack) == 0
    
    def size(self):
        return len(self.stack)
    
myStack = Stack()

'''myStack.push(1)
myStack.push(2)
myStack.push(3)
myStack.push(4)
myStack.push(5)
myStack.push(6)
myStack.push(7)
myStack.push('A')'''

'''print("Stack: ", myStack.stack)
print("Pop: ", myStack.pop())
print("Stack after pop: ", myStack.stack)
print("Peek: ", myStack.peek())
print("isEmpty: ", myStack.isEmpty())
print("Size: ", myStack.size())'''

File: 2/Programozási nyelvek/gyakorlat/homework4/Stack.java
package homework4;

import java.util.ArrayList;
import java.util.List;

public class Stack<T> {
    private final List<T> stack;

    public Stack() {
        this.stack = new ArrayList<>();
    }

    public void push(T item) {
        stack.add(item);
    }

    public T pop() throws IllegalStateException {
        if (empty()) {
            throw new IllegalStateException("Cannot pop from an empty stack");
        }
        return stack.remove(stack.size() - 1);
    }

    public T peek() throws IllegalStateException {
        if (empty()) {
            throw new IllegalStateException("Cannot peek an empty stack");
        }
        return stack.get(stack.size() - 1);
    }

    public boolean empty() {
        return stack.isEmpty();
    }
}